docs/internals/contributing/writing-code/coding-style.txt

```diff
@@ -4,6 +4,32 @@ Coding style
 
 Please follow these coding standards when writing code for inclusion in Django.
 
+.. _coding-style-pre-commit:
+
+Pre-commit checks
+=================
+
+`pre-commit <https://pre-commit.com>`_ is a framework for managing pre-commit
+hooks. These hooks help to identify simple issues before committing code for
+review. By checking for these issues before code review it allows the reviewer
+to focus on the change itself, and it can also help to reduce the number CI
+runs.
+
+To use the tool, first install ``pre-commit`` and then the git hooks::
+
+.. console::
+
+    $ pip install pre-commit
```

Comment:
Sorry that I missed this during the previous rounds, but since ticket-30367 this should be `python -m pip install pre-commit`. Fixed in #13804.

Reply:
Yes, I've suggested this today, see https://github.com/django/django/pull/13696#discussion_r547859981.

Reply:
Ah, yes, I missed that bit. (Is it Christmas yet? 🎅) 

(I know this is the official line, and what we recommend, but I wonder, do you really type that? I don't 🙂)

Reply:
I use `pipx run pre-commit run --all-files`

Reply:
I have `alias pip=python -m pip`

Reply:
See both of those solutions are very clever. 😄

Reply:
I personally do not. I have pre-commit installed via the Fedora `dnf` package manager.

However, I don't consider this instruction targeted at me and you, but new contributors and perhaps contributors that have only a limited amount of Python experience but are keen nonetheless.

The reason Django suggests using `python -m pip install ...` is to avoid the wrong `pip` environment from executing, leading to confusion. This confusion inevitably leads to more support question in one of the many support channels or internally among teams. If this can be handled earlier by the docs, it is a helpful to all, the new contributors and the mentors. I think the original justification still applies.

In the end, I think how the docs instruct users to install Python packages should be reasonably consistent. If `pip install` is good enough here, why not in other articles?

Reply:
No, I agree @jdufresne. I was just wondering. (Fully onboard with that being the recommended incantation.) 👍